src/config.ts

```diff
@@ -13,1213 +13,23 @@
-
-  /**
-   * The player relies on browser events and properties to update its status to
-   * "ENDED".
-   *
-   * Sadly in some cases, like in Chrome 54, this event is never triggered on
-   * some contents probably due to a browser bug.
-   *
-   * This threshold resolves this issue by forcing the status to "ENDED" when:
-   *   1. the player is stalling
-   *   2. the absolute difference between current playback time and duration is
-   *      under this value
-   *
-   * If set to null, this workaround is disabled and the player only relies on
-   * browser events.
-   *
-   * @type {Number|null}
-   */
-  FORCED_ENDED_THRESHOLD: 0.0008,
-
-  /**
-   * Maximum duration from the current position we will let in the buffer when
-   * switching an Adaptation of a given type.
-   *
-   * For example, if we have ``text: { before: 1, after: 4 }``, it means that
-   * when switching subtitles, we will let 1 second before and 4 second after
-   * the current position in the previous language (until the new segments
-   * overwrite it).
-   * This is to allow smooth transitions and avoid de-synchronization that
-   * can happen when removing the content being decoded.
-   * @type {Object}
-   */
-  ADAPTATION_SWITCH_BUFFER_PADDINGS: {
-    video: { before: 2, after: 2.5 },
-    audio: { before: 2, after: 2.5 },
-    text: { before: 0, after: 0 }, // not managed natively, so no problem here
-    image: { before: 0, after: 0 }, // not managed natively, so no problem here
-  },
-
-  /**
-   * Interval, in milliseconds, at which we should manually flush
-   * SourceBuffers.
-   * Some browsers (happened with firefox 66) sometimes "forget" to send us
-   * `update` or `updateend` events.
-   * In that case, we're completely unable to continue the queue here and
-   * stay locked in a waiting state.
-   * This interval is here to check at regular intervals if the underlying
-   * SourceBuffer is currently updating.
-   * @type {Number}
-   */
-  SOURCE_BUFFER_FLUSHING_INTERVAL: 500,
-
-  /**
-   * Any already-pushed segment starting before or at the current position +
-   * CONTENT_REPLACEMENT_PADDING won't be replaced by new segments.
-   *
-   * This allows to avoid overwriting segments that are currently being decoded
-   * as we encountered many decoding issues when doing so.
-   * @type {Number} - in seconds
-   */
-  CONTENT_REPLACEMENT_PADDING: 1.2,
 
-  /**
-   * For video and audio segments, determines two thresholds below which :
-   * - The segment is considered as loaded from cache
-   * - The segment may be loaded from cache depending on the previous request
-   */
-  CACHE_LOAD_DURATION_THRESHOLDS: {
-    video: 50,
-    audio: 10,
-  },
+class ConfigHandler {
+  _config = DEFAULT_CONFIG;
 
-  /** Interval we will use to poll for checking if an event shall be emitted */
-  STREAM_EVENT_EMITTER_POLL_INTERVAL: 250,
+  update(config: Partial<IDefaultConfig>) {
+    const newConfig = deepMerge(DEFAULT_CONFIG, config) as IDefaultConfig;
```

Comment:
Why taking the `DEFAULT_CONFIG` as a base again?

Shouldn't that be more something like:
```ts
deepMerge(this._config, config)
```

Reply:
Yes, just thought about this too, will be changed

Reply:
Fixed